hcl/ast/expression_test.go

```diff
@@ -92,6 +92,69 @@ a b c
 EOT
 `,
 		},
+		{
+			name: "heredocs have unespaced quotes",
+			expr: `"name=\"val\"\n"`,
+			want: `<<-EOT
+name="val"
+EOT
+`,
+		},
+		{
+			name: "heredocs have unescaped backslashes",
+			expr: `"\\\n"`,
+			want: `<<-EOT
+\
+EOT
+`,
+		},
+		{
+			name: "heredocs with mixed escaped symbols",
+			expr: `"\n\\\n\\\n"`,
+			want: `<<-EOT
+
+\
+\
+EOT
+`,
+		},
+		{
+			name: "carriege returns generates plain strings",
```

Comment:
please add test for escaped `\n` at EOS: `"a\\n"`

Reply:
added just in case, but there were tests covering this already: https://github.com/mineiros-io/terramate/pull/841/commits/5cd85ac48d03d4b999b47a6ae2e3cf6bc46f5f9f

thanks for testing!